components/properties/src/provider.rs

```diff
@@ -311,11 +314,128 @@ impl<'data, T: TrieValue> PropertyCodePointMapV1<'data, T> {
     }
 }
 
+/// `NormalizedPropertyNameBorrowed` is a fake type enabling us to use `#[zerovec::make_varule]`
+/// conveniently.
+/// This is a property name that can be "loose matched" as according to
+/// https://www.unicode.org/Public/UCD/latest/ucd/PropertyValueAliases.txt
+///
+/// (matched case-insensitively in ASCII, ignoring underscores, whitespace, and hyphens)
+///
+/// This is expected to be ASCII, but we do not rely on this invariant anywhere except during
+/// datagen.
+#[derive(PartialEq, Eq)] // VarULE wants these to be byte equality
+#[zerovec::make_varule(NormalizedPropertyName)]
```

Comment:
Nit: I prefer DSTs to be clearly indicated, such as by having "str" or "ULE" in their name.

Suggestion: `NormalizedPropertyNameString` that turns into `NormalizedPropertyNameStr`. Or, just avoid having two types and only have the ULE type, like for UnvalidatedStr.

Reply:
"just the ULE type" means I have to do a whole bunch of manual impl work

we should add something to zerovec_derive that can do all the stuff `make_varule` does without needing a source ULE type

Reply:
I'll just call it Str and String (or StringUnused)

Reply:
i realized that we can't call it String since it's a borrowed type. How about Borrowed and Str?

Reply:
Is there really _that much_ manual work if you just implement VarULE directly? Is your goal to eliminate `unsafe` code? (I've never been _super_ happy with the derive-generated type; I'd prefer a derive that generates impls but not actual types)

Reply:
Yeah so I started with it and there was just a ton of boilerplate around serde etc.

It's not actually unsafe (otherwise I would have loudly griped about it 😝)

But now that I have it all working I can definitely try again.

Reply:
Yeah, I think it's worth it to avoid the two types. I would like to see derives that generate these same impls on types that are variations of `[u8]` (or `[u8; N]`) since we have a few of them now. For most of these types we can use the ULE types directly and we don't need the concrete types to be different.

Here's the one for UnvalidatedStr; it can probably mostly be copied:

https://github.com/unicode-org/icu4x/blob/084350ed6a48cdd327dcca6068cb38c1fc2f8693/utils/zerovec/src/ule/unvalidated.rs#L131

Reply:
and I realize there's already an issue for that: https://github.com/unicode-org/icu4x/issues/2310